compat/wserrno.cc

```diff
@@ -19,27 +19,22 @@ SetErrnoFromWsaError()
     // POSIX codes which socket API users may care about
     static const auto *CodeMap = new std::unordered_map<int, int> {
 
-        // values checked for by accept(2) callers
         { WSAECONNABORTED, ECONNABORTED },
 
-        // values checked for by connect(2) callers
         { WSAEINPROGRESS, EINPROGRESS },
         { WSAEAFNOSUPPORT, EAFNOSUPPORT },
         { WSAEINVAL, EINVAL },
         { WSAEISCONN, EISCONN },
 
-        // values checked by ignoreErrno()
         { WSAEWOULDBLOCK, EWOULDBLOCK },
-        // WSAEAGAIN not defined
+        // WSAEAGAIN has no equivalent
         { WSAEALREADY, EALREADY },
         { WSAEINTR, EINTR },
-        // WSARESTART not defined
+        // WSARESTART has no equivalent
```

Comment:
Hm... We use ERESTART (inside `#ifdef`). However, Google AI thinks that "WSARESTART" is itself a "typo or misunderstanding", and I cannot find that constant in Windows documentation.

```suggestion
```

Or perhaps you meant WSATRY_AGAIN? Windows API [description](https://learn.microsoft.com/en-us/windows/win32/winsock/windows-sockets-error-codes-2) of WSATRY_AGAIN does not seem to match, but in case it should be used anyway, we can try something like the suggestion below.

```suggestion
#if defined(ERESTART)
        { WSATRY_AGAIN, ERESTART },
#else
        { WSATRY_AGAIN, EINTR },
#endif
```

Reply:
Yes. `WSARESTART` does not exist. Same should apply here as for the `ENFILE` and `EAGAIN` cases.

FWIW, The code using `ERESTART` is `ignoreErrno()` - Specifically ... ignoring a code that does not exist. No code needed.

If there are `WSA...` (not `WSAE...`) codes that `ignoreErrno()` function or other code should match they should be listed in that function explicitly, using a `_SQUID_WINDOWS_||_SQUID_MINGW_` wrapper. Not remapped to a semantically different code before use by the wider Squid code.

Reply:
> Yes. `WSARESTART` does not exist. 

AFAICT, my first suggestion applies then. Please do not list/mention/etc. made-up error codes! If you insist on doing so, I will insist on a C++ comment explaining that/why we are making stuff up so that code readers do not waste time searching for error codes that do not exist.

If it is accurate, we can also add a comment like this if you prefer:

```suggestion
        // no Windows error code maps to ERESTART
```


> If there are `WSA...` (not `WSAE...`) codes that `ignoreErrno()` function or other code should match they should be listed in that function explicitly, using a `_SQUID_WINDOWS_||_SQUID_MINGW_` wrapper. Not remapped to a semantically different code before use by the wider Squid code.

I agree that we should not map to a semantically different code. AFAICT, the correctness of the "If there are..." sentence quoted above depends on whether a semantically-same mapping exists for those Windows error codes:

* If a Windows error code has a semantically equivalent POSIX error code used by Squid, we should map.
* If a Windows error code has a semantically equivalent POSIX error code currently unused by Squid, we may map, but do not have to.
* In all other cases, we should not map.